src/components/molecules/ToolCard.tsx

```
@@ -45,6 +60,325 @@ const ToolCard = ({ id, tool_results, note_id, note_title, clear, updateToolResu
+                    const tool = tools.find((tool) => tool.name === c.name);
+                    if (!tool) return a;
+                    if (tool.auto_approve.includes(c.method)) {
+                        return a;
+                    }
+
+                    return tool.ai_auto_approve ? [...a, c] : a;
+                }, [] as Cmd[]);
+            })();
+
+            const aiAutoApprove = await (async () => {
+                if (cmdsRequiredAiAutoApprove.length === 0) return false;
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)